skbio/sequence/_sequence.py

```diff
@@ -503,43 +519,8 @@ def __init__(self, sequence, metadata=None,
                 positional_metadata = sequence.positional_metadata
             sequence = sequence._bytes
```

Comment:
I was gonna ask about this. Went ahead and added it but would like to discuss further at some point.